ocaml/xapi/xapi_xenops.ml

```diff
@@ -2370,87 +2370,63 @@ let events_from_xapi () =
-                then error "events_from_xapi: missing from the cache: [ %s ]" (String.concat "; " missing_in_cache);
-                if extra_in_cache <> []
-                then error "events_from_xapi: extra items in the cache: [ %s ]" (String.concat "; " extra_in_cache);
-
-                let classes = List.map (fun x -> Printf.sprintf "VM/%s" (Ref.string_of x)) resident_VMs in
-                (* NB we re-use the old token so we don't get events we've already
-                   							   received BUT we will not necessarily receive events for the new VMs *)
-
-                while true do
-                  let api_timeout = 60. in
-                  let timeout = 30. +. api_timeout +. !Db_globs.master_connection_reset_timeout in
-                  let timebox_rpc = Helpers.make_timeboxed_rpc ~__context timeout in
-                  let from =
-                    try
-                      XenAPI.Event.from
-                        ~rpc:timebox_rpc
-                        ~session_id ~classes
-                        ~token:!token
-                        ~timeout:api_timeout
-                      |> event_from_of_rpc
-                    with e ->
-                      Debug.log_backtrace e (Backtrace.get e);
-                      raise e
-                  in
-                  if List.length from.events > 200 then warn "Warning: received more than 200 events!";
-                  List.iter
-                    (function
-                      | { ty = "vm"; reference = vm' } ->
-                        let vm = Ref.of_string vm' in
-                        begin
-                          try
-                            let id = id_of_vm ~__context ~self:vm in
-                            let resident_here = Db.VM.get_resident_on ~__context ~self:vm = localhost in
-                            debug "Event on VM %s; resident_here = %b" id resident_here;
-                            if resident_here
-                            then Xenopsd_metadata.update ~__context ~self:vm |> ignore
-                          with e ->
-                            if not(Db.is_valid_ref __context vm)
-                            then debug "VM %s has been removed: event on it will be ignored" (Ref.string_of vm)
-                            else begin
-                              error "Caught %s while processing XenAPI event for VM %s" (Printexc.to_string e) (Ref.string_of vm);
-                              raise e
-                            end
-                        end
-                      | _ -> warn "Received event for something we didn't register for!"
-                    ) from.events;
-                  token := from.token;
-                  Events_from_xapi.broadcast !token;
-                done
-             )
-         with
-         | Api_errors.Server_error(code, _) when code = Api_errors.session_invalid ->
-           debug "Woken event thread: updating list of event subscriptions"
-         | e ->
```

Comment:
We don't need the exception case for `session_invalid` anymore, but we do need the catch-all one, otherwise we risk exiting from the event loop.